tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -646,9 +661,47 @@ bpf_code_generator::emit_c_code(std::ostream& output_stream)
     }
 
     for (auto& [name, section] : sections) {
-        auto function_name = !section.function_name.empty() ? section.function_name : name;
+        auto program_name = !section.program_name.empty() ? section.program_name : name;
+
+        // Emit the program and attach type GUID.
+        std::string program_type_name = program_name + "_program_type_guid";
+        std::string attach_type_name = program_name + "_attach_type_guid";
+
+        output_stream << format_string(
+                             "static GUID %s = %s;",
```

Comment:
```suggestion
                             "static ebpf_program_type_t %s = %s;",
```

Reply:
bpf_code_generator.cpp is compiled on Linux also (as part of bpf2c-test in CI/CD). So this file cannot include Windows specific headers. Including ebpfapi.h to this file causes issues.

Reply:
I didn't follow why this is relevant to Linux, can you elaborate?

Reply:
@Alan-Jowett can share more details, but IIUC `bpf2c_tests` tests the sample programs in ubpf repo. To do that it uses Python, but Windows version of Python does not work correctly and Alan was able to make it work on Linux. This is the reason this file needs to be compiled on Linux also (to be able to test the sample programs in ubpf repo using bpf2c).

Reply:
Filed #870 to track removing dependency on Linux build.